src/clang.rs

```diff
@@ -136,6 +136,19 @@ impl Cursor {
         self.specialized().is_valid()
     }
 
+    pub fn is_fully_specialized_template(&self) -> bool {
+        self.is_template() && self.num_template_args() > 0
+    }
+
+    pub fn is_in_non_fully_specialized_template(&self) -> bool {
+        if self.is_toplevel() {
+            return false;
+        }
+        let parent = self.semantic_parent();
+        (parent.is_template() && !parent.is_fully_specialized_template()) ||
```

Comment:
You can drop the `parent.is_template()` check here, since is already part of `is_fully_specialized_template()`.